MuColl/MAIA/compact/MAIA_v0/HCalEndcap_o1_v01_01.xml

```diff
@@ -0,0 +1,163 @@
+		<!--constant name="HCalRing_layers" value=" floor( (HCalEndcap_min_z - HCalRing_min_z) / HCalEndcap_layer_thickness + 0.5)"/-->
+		<constant name="HCalStep_layers" value="HCalEndcap_layers/HCalEndcap_inner_steps"/>
+		<constant name="HCalStep_dr" value="(HCalEndcap_inner_radius2 - HCalEndcap_inner_radius)/HCalEndcap_inner_steps"/>
+	</define>
+
+	<detectors>
+		<detector name="HCalEndcaps" type="DD4hep_SubdetectorAssembly" vis="HCALVis">
+				<shape type="BooleanShape" operation="Subtraction" material="Air">
+					<shape type="BooleanShape" operation="Subtraction">
+						<shape type="BooleanShape" operation="Subtraction">
+							<shape type="BooleanShape" operation="Subtraction">
+								<!--shape type="PolyhedraRegular"  numsides="HCalEndcap_symmetry" rmin="0" rmax="HCalEndcap_outer_radius+env_safety" dz="2.0*HCalEndcap_max_z+2.0*env_safety"/>
+								<shape type="PolyhedraRegular"  numsides="HCalEndcap_symmetry" rmin="0" rmax="HCalRing_inner_radius-env_safety" dz="2.0*HCalEndcap_min_z-2.0*env_safety"/-->
+								<shape type="PolyhedraRegular"  numsides="HCalEndcap_symmetry" rmin="HCalEndcap_inner_radius-env_safety" rmax="HCalEndcap_outer_radius+10.0*env_safety" dz="2.0*HCalEndcap_max_z+2*env_safety"/>
+								<shape type="PolyhedraRegular"  numsides="HCalEndcap_symmetry" rmin="0" rmax="HCalEndcap_outer_radius+10.0*env_safety" dz="2.0*HCalEndcap_min_z-2*env_safety"/>
+							</shape>
+							<!--shape type="PolyhedraRegular"  numsides="HCalEndcap_symmetry" rmin="0" rmax="HCalEndcap_outer_radius+2.0*env_safety" dz="2.0*HCalRing_min_z-2.0*env_safety"/-->
+						</shape>
+					</shape>
+					<shape type="Polycone" startphi="0" deltaphi="2.0*pi*rad">
+						<zplane z="-HCalEndcap_max_z-env_safety" rmin="0" rmax="HCalEndcap_inner_radius2"/>
+						<zplane z="-HCalEndcap_min_z"            rmin="0" rmax="HCalEndcap_inner_radius"/>
+						<zplane z="HCalEndcap_min_z"             rmin="0" rmax="HCalEndcap_inner_radius"/>
+						<zplane z="HCalEndcap_max_z+env_safety"  rmin="0" rmax="HCalEndcap_inner_radius2"/>
+					</shape>
+				</shape>
+				<rotation x="0*deg" y="0*deg" z="90*deg-180*deg/HCalEndcap_symmetry"/>
+
+			<comment>HCalEndcap Assembly</comment>
+			<composite name="HCalEndcap"/>
+			<!--composite name="HCalRing"/-->
+		</detector>
+	</detectors>
+
+
+	<!--  Definition of the readout segmentation  -->
+	<readouts>
+		<readout name="HCalEndcapCollection">
+			<segmentation type="CartesianGridXY" grid_size_x="HCal_cell_size" grid_size_y="HCal_cell_size" />
+			<id>${GlobalCalorimeterReadoutID}</id>
+		</readout>
+		<!--readout name="HCalRingCollection">
+			<segmentation type="CartesianGridXY" grid_size_x="HCal_cell_size" grid_size_y="HCal_cell_size" />
+			<id>${GlobalCalorimeterReadoutID}</id>
+		</readout-->
+	</readouts>
+
+
+	<detectors>
+
+		<detector name="HCalEndcap" type="GenericCalEndcap_o2_v01" id="DetID_HCAL_Endcap" readout="HCalEndcapCollection" vis="HCALVis" >
```

Comment:
This is where the discussion connects to https://github.com/key4hep/k4geo/pull/341